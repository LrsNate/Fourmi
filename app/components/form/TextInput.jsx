import TextField from 'material-ui/TextField';
import React, { Component } from 'react';
import { FormField } from 'react-form';

type Props = {
  field: string,
  label: string
};

class TextInput extends Component<Props> {
  render() {
    const { field, label } = this.props;

    return (
      <FormField field={field}>
        {({ getValue, setValue, setTouched }) =>
          <TextField
            label={label}
            value={getValue()}
            onChange={(e: string) => setValue(e)}
            onBlur={setTouched}
            fullWidth
          />}
      </FormField>
    );
  }
}

export default TextInput;
