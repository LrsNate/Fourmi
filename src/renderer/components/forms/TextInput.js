import { TextField } from "material-ui";
import PropTypes from "prop-types";
import React from "react";
import { Field } from "react-final-form";

const TextInput = ({ name, ...props }) => {
  return (
    <Field name={name}>
      {({ input }) => (
        <TextField {...input} {...props} color="accent" margin="normal" />
      )}
    </Field>
  );
};

TextInput.propTypes = {
  name: PropTypes.string.isRequired
};

export default TextInput;
