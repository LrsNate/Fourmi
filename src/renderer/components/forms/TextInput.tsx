import { TextField } from "@material-ui/core";
import React from "react";
import { Field, FieldRenderProps } from "react-final-form";

interface TextInputProps {
  name: string;
  label: string;
}

const TextInput = ({ name, ...props }: TextInputProps) => {
  return (
    <Field name={name}>
      {({ input }: FieldRenderProps) => (
        <TextField {...input} {...props} fullWidth={true} margin="normal" />
      )}
    </Field>
  );
};

export default TextInput;
