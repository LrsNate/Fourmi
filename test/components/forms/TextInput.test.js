import { render } from "enzyme";
import React from "react";
import { Form } from "react-final-form";

import TextInput from "../../../src/renderer/components/forms/TextInput";

describe("The TextInput component", () => {
  describe("On an initial load", () => {
    it("matches the snapshot", () => {
      const wrapper = render(
        <Form onSubmit={() => {}}>
          {({ handleSubmit }) => (
            <form onSubmit={handleSubmit}>
              <TextInput label="Some label" name="someName" />
            </form>
          )}
        </Form>
      );

      expect(wrapper).toMatchSnapshot();
    });
  });
});
