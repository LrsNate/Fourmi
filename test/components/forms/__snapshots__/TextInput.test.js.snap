// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The TextInput component On an initial load matches the snapshot 1`] = `
<form>
  <div
    class="MuiFormControl-root-1 MuiFormControl-marginNormal-2"
    color="accent"
  >
    <label
      class="MuiFormLabel-root-11 MuiInputLabel-root-5 MuiInputLabel-formControl-6 MuiInputLabel-animated-9"
      data-shrink="false"
    >
      Some label
    </label>
    <div
      class="MuiInput-root-15 MuiInput-formControl-16 MuiInput-inkbar-17 MuiInput-underline-21"
    >
      <input
        class="MuiInput-input-24 MuiInput-inputSingleline-27"
        name="someName"
        type="text"
        value=""
      />
    </div>
  </div>
</form>
`;
